src/intrinsics.cpp

```diff
@@ -499,7 +499,7 @@ static jl_cgval_t generic_bitcast(jl_codectx_t &ctx, const jl_cgval_t *argv)
         return mark_julia_type(ctx, vx, false, bt);
     }
     else {
-        Value *box = emit_allocobj(ctx, nb, boxed(ctx, bt_value));
+        Value *box = emit_allocobj(ctx, nb, literal_pointer_val(ctx, (jl_value_t*)bt));
```

Comment:
`bt` isn't a type of the object (it's a partial supertype)

Reply:
Reverted. I thought the primitive check above filters out ones with free parameters but seems not (and I missed the branch right above).

The change was during one of my attempt to eliminate one of the last few maybe's which I subsequently gave up so it's not necessary here....